### ISSUE TITLE
consider Windows edge case

### ISSUE THREAD

**Issue by borkdude** [babashka/process#2]
See
- https://github.com/borkdude/babashka.curl/blob/master/src/babashka/curl.clj#L20
- https://github.com/lread/rewrite-cljc-playground/blob/3aa4741a773ccd329c350208c50a83b1e8ce9671/script/helper/shell.clj#L21
- https://github.com/borkdude/clj-kondo/blob/de925fa93ede2672cba5de4d36cfd3c21e0a7f72/test/clj_kondo/test_utils.clj#L110

**Comment by lread** [comment]
It seems the escape argument algorithm for Windows might be a smidge more complicated than we had guessed.

Someone worked out an algorithm here: https://bugs.java.com/bugdatabase/view_bug.do?bug_id=6511002

And I am not sure if carets `^` are an issue, they were causing me grief in my bb scripts with my own ProcessBuilder code:
`"{:output {:include-files [\"^src\" \"^test\" \"^script\"]}}"`. 
I simply moved this config from the command line to a file to avoid figuring out what I needed to do here.
Worth testing carets when we get to this issue.

**Comment by borkdude** [comment]
Repro:

This works on Windows, but on macOS we don't need the quoting:

```
(-> ($ clojure -Spath -Sdeps "{:deps {clj-kondo/clj-kondo {:mvn/version \\\"RELEASE\\\"}}}") :out slurp print)
```

**Comment by lread** [comment]
I'm doing a few experiments to learn more about how escaping works on Java/Windows.

So far I see that `ProcessBuilder` allows the executable to be an `.exe`, `.bat` or `.cmd`.  It does not seem to accept a `.ps1` script directly though.

Escaping rules seems to differ when your executable is an `.exe` vs `.bat`/`.cmd`.

I'll setup a GitHub repo playground sometime soon.

**Comment by borkdude** [comment]
@lread I guess people can do `($ powershell.exe foo.ps1)` right

**Comment by lread** [comment]
> @lread I guess people can do `($ powershell.exe foo.ps1)` right

Yup!

**Comment by borkdude** [comment]
Addressed with cd04e1b9e178cab7994b4a0472c71f0885ce03d5. `*escape-fn*` can be bound/altered if this isn't a good default.
